cmd/bootstrapper/download/client.go

```diff
@@ -0,0 +1,97 @@
+package download
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/Dynatrace/dynatrace-bootstrapper/pkg/configure/oneagent/ca"
+	dtclient "github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer/url"
+	"github.com/spf13/afero"
+)
+
+type InstallerBuilder func(fs afero.Fs, dtc dtclient.Client, props *url.Properties) installer.Installer
+type DTClientBuilder func(url, apiToken, paasToken string, opts ...dtclient.Option) (dtclient.Client, error)
+
+type Client struct {
+	newInstaller InstallerBuilder
+	newDTClient  DTClientBuilder
+}
+
+type Option func(*Client)
+
+func WithInstaller(builder InstallerBuilder) Option {
+	return func(cl *Client) {
+		cl.newInstaller = builder
+	}
+}
+
+func WithDTClient(builder DTClientBuilder) Option {
+	return func(cl *Client) {
+		cl.newDTClient = builder
+	}
+}
+
+func New(options ...Option) *Client {
+	cl := &Client{
+		newInstaller: url.NewUrlInstaller,
+		newDTClient:  dtclient.NewClient,
+	}
+
+	for _, opt := range options {
+		opt(cl)
+	}
+
+	return cl
+}
+
+func (cl *Client) Do(ctx context.Context, fs afero.Afero, inputDir string, targetDir string, props url.Properties) error {
+	client, err := cl.createClientFromFs(fs, inputDir)
+	if err != nil {
+		return err
+	}
+
+	oneAgentInstaller := cl.newInstaller(
+		fs,
+		client,
+		&props,
+	)
+
+	_, err = oneAgentInstaller.InstallAgent(ctx, targetDir)
+
+	return err
+}
+
+func (cl *Client) createClientFromFs(fs afero.Afero, inputDir string) (dtclient.Client, error) {
```

Comment:
this method name a bit confusing to me, because it's method of Client struct and it returns dtclient.Client, so IMO better to name it: `createDTClient`. Regarding "from FS" ( IMO arguments already explaining to us that it's from FS) but I'm okey if we keep it as is: `createDTClientFromFS`